### ISSUE TITLE
tcpreplay fails to open already allocated tap0 device

### ISSUE THREAD

**Issue by rmstoi** [appneta/tcpreplay#411]
Hi,

I've been using tcpreplay with the Zephyr Project RTOS
https://github.com/zephyrproject-rtos/zephyr

tcpreplay version 4, provided from the package repositories (Debian, Ubuntu, Fedora)  does not work anymore with the TAP interface. The last working version is tcpreplay 3.4.4.

The problem is that tun network device allocation (/dev/net/tun) is done by tunslip6.c in Zephyr net-tools before tcpreplay.

https://github.com/zephyrproject-rtos/net-tools/blob/master/tunslip6.c#L720

So when tcpreplay is ran, it tries to also open tun device and reports:

```
tcpreplay -i tap0 packet.pcapng

Fatal Error in sendpacket.c:sendpacket_open() line 539:
 failed to open device tap0: Unable to create tuntap interface: tap0

```

I can workaround this by recompiling tcpreplay with:

`./configure --disable-tuntap`

but it would be nice if tcpreplay from the package repository works.

Maybe tcpreplay could have a command line option to open /dev/net/tun?
Otherwise it would use already opened device.

Thanks!

**Comment by fklassen** [comment]
Will investigate for next release

**Comment by whitequark** [comment]
Simpler workaround: rename the device from `tap0` to e.g. `xxx0` (or anything that doesn't start with `tap`).

**Comment by fklassen** [comment]
@whitequark that explains why my testing never found anything wrong. I will keep the default name `tap0`.

This is legacy code, and we may be better of with it deleted. I'll have a look.

**Comment by fklassen** [comment]
fixed in PR #504

**Comment by jersey99** [comment]
@fklassen FYI, I am using 4.3.3-2 and I still had to use @whitequark 's workaround (basically renaming the device)

**Comment by fklassen** [comment]
> @fklassen FYI, I am using 4.3.3-2 and I still had to use @whitequark 's workaround (basically renaming the device)

Opened #651 to address. Please fill in system information into this new issue.

**Comment by fklassen** [comment]
Fixed in PR #660
